internal/fs/vss_windows.go

```diff
@@ -0,0 +1,1054 @@
+	getWriterStatus               uintptr
+	setBackupSucceeded            uintptr
+	setBackupOptions              uintptr
+	setSelectedForRestore         uintptr
+	setRestoreOptions             uintptr
+	setAdditionalRestores         uintptr
+	setPreviousBackupStamp        uintptr
+	saveAsXML                     uintptr
+	backupComplete                uintptr
+	addAlternativeLocationMapping uintptr
+	addRestoreSubcomponent        uintptr
+	setFileRestoreStatus          uintptr
+	addNewTarget                  uintptr
+	setRangesFilePath             uintptr
+	preRestore                    uintptr
+	postRestore                   uintptr
+	setContext                    uintptr
+	startSnapshotSet              uintptr
+	addToSnapshotSet              uintptr
+	doSnapshotSet                 uintptr
+	deleteSnapshots               uintptr
+	importSnapshots               uintptr
+	breakSnapshotSet              uintptr
+	getSnapshotProperties         uintptr
+	query                         uintptr
+	isVolumeSupported             uintptr
+	disableWriterClasses          uintptr
+	enableWriterClasses           uintptr
+	disableWriterInstances        uintptr
+	exposeSnapshot                uintptr
+	revertToSnapshot              uintptr
+	queryRevertStatus             uintptr
+}
+
+// getVTable returns the vtable for IVssBackupComponents.
+func (vss *IVssBackupComponents) getVTable() *IVssBackupComponentsVTable {
+	return (*IVssBackupComponentsVTable)(unsafe.Pointer(vss.RawVTable))
+}
+
+// InitializeForBackup calls the equivalent VSS api.
+func (vss *IVssBackupComponents) InitializeForBackup() HRESULT {
+	result, _, _ := syscall.Syscall(
+		vss.getVTable().initializeForBackup, 2,
+		uintptr(unsafe.Pointer(vss)),
+		0, 0,
+	)
+
+	return HRESULT(result)
+}
+
+// SetContext calls the equivalent VSS api.
+func (vss *IVssBackupComponents) SetContext(context VssContext) HRESULT {
+	result, _, _ := syscall.Syscall(
+		vss.getVTable().setContext, 2,
+		uintptr(unsafe.Pointer(vss)),
+		uintptr(context),
+		0,
+	)
+
+	return HRESULT(result)
+}
+
+// GatherWriterMetadata calls the equivalent VSS api.
+func (vss *IVssBackupComponents) GatherWriterMetadata() (HRESULT, *IVSSAsync) {
+	var oleIUnknown *ole.IUnknown
+
+	result, _, _ := syscall.Syscall(
+		vss.getVTable().gatherWriterMetadata, 2,
+		uintptr(unsafe.Pointer(vss)),
+		uintptr(unsafe.Pointer(&oleIUnknown)),
+		0,
+	)
+
+	return vss.handleAsyncReturnValue(result, oleIUnknown)
+}
+
+// IsVolumeSupported calls the equivalent VSS api.
+func (vss *IVssBackupComponents) IsVolumeSupported(
+	volumeName string,
+) (HRESULT, bool) {
+	volumeNamePointer, _ := syscall.UTF16PtrFromString(volumeName)
+	isSupported := false
+
+	var result uintptr = 0
+
+	if runtime.GOARCH == "386" {
+		result, _, _ = syscall.Syscall9(
+			vss.getVTable().isVolumeSupported, 7,
+			uintptr(unsafe.Pointer(vss)),
+			uintptr(ole.IID_NULL.Data1),
+			uintptr(ole.IID_NULL.Data2),
+			uintptr(ole.IID_NULL.Data3),
+			uintptr(ole.IID_NULL.Data4[0]),
```

Comment:
Fixed in https://github.com/restic/restic/pull/2274/commits/d6eb1bbb56f24f12dacd9dbb95f8b0d8aaa83013

Reply:
This is _not_ the pattern from `DeleteSnapshots`. The 128bit uuid should be passed via _four_ 32-bit register not five, so this will break in mysterious ways.